File: upcoming/source/includes/write/replace.c
#include <stdio.h>
#include <bson/bson.h>
#include <mongoc/mongoc.h>

int
main (void)
{
    mongoc_client_t *client;
    mongoc_collection_t *collection;
    mongoc_init ();

    client = mongoc_client_new ("<connection string URI>");
    collection = mongoc_client_get_collection (client, "sample_restaurants", "restaurants");
    
    {
        // start-replace-one
        bson_t *query = BCON_NEW ("name", "Pizza Town");
        bson_t *replace = BCON_NEW (
            "name", "Mongo's Pizza",
            "cuisine", "Pizza",
            "address", "{",
            "street", "123 Pizza St",
            "zipCode", "10003",
            "}",
            "borough", "Manhattan"
        );
        bson_error_t error;

        if (!mongoc_collection_replace_one (collection, query, replace, NULL, NULL, &error)) {
            fprintf (stderr, "Replace operation failed: %s\n", error.message);
        }

        bson_destroy (query);
        bson_destroy (replace);
        // end-replace-one
    }

    {
        // start-replace-options
        bson_t *query = BCON_NEW ("name", "Food Town");
        bson_t *replace = BCON_NEW (
            "name", "Food World",
            "cuisine", "Mixed",
            "address", "{",
            "street", "123 Food St",
            "zipCode", "10003",
            "}",
            "borough", "Manhattan"
        );
        bson_error_t error;

        bson_t opts;
        bson_init (&opts);
        bson_append_bool (&opts, "upsert", -1, true);

        if (!mongoc_collection_replace_one (collection, query, replace, &opts, NULL, &error)) {
            fprintf (stderr, "Replace operation failed: %s\n", error.message);
        } 

        bson_destroy (query);
        bson_destroy (replace);
        bson_destroy (&opts);
        // end-replace-options
    }

    mongoc_collection_destroy (collection);
    mongoc_client_destroy (client);
    mongoc_cleanup ();

    return EXIT_SUCCESS;
}

File: upcoming/source/includes/write/replace.php
<?php
require 'vendor/autoload.php';

$uri = getenv('MONGODB_URI') ?: throw new RuntimeException('Set the MONGODB_URI variable to your Atlas URI that connects to the sample dataset');
$client = new MongoDB\Client($uri);

// start-db-coll
$collection = $client->sample_restaurants->restaurants;
// end-db-coll

// start-replace-one
$replaceDocument = [
    'name' => 'Mongo\'s Pizza',
    'cuisine' => 'Pizza',
    'address' => [
        'street' => '123 Pizza St',
        'zipCode' => '10003',
    ],
    'borough' => 'Manhattan',
];

$result = $collection->replaceOne(['name' => 'Pizza Town'], $replaceDocument);
echo 'Modified documents: ', $result->getModifiedCount();
// end-replace-one

// start-replace-options
$replaceDocument = [
    'name' => 'Food World',
    'cuisine' => 'Mixed',
    'address' => [
        'street' => '123 Food St',
        'zipCode' => '10003',
    ],
    'borough' => 'Manhattan',
];

$result = $collection->replaceOne(
    ['name' => 'Food Town'],
    $replaceDocument,
    ['upsert' => true],
);
// end-replace-options
